From 45699f3ca10e3f2eae5187cbd0a9c27440532351 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:33 -0400
Subject: [powerpc] Remove non-uapi linkage.h export

Message-id: <1410545655-205645-525-git-send-email-dzickus@redhat.com>
Patchwork-id: 94509
O-Subject: [RHEL7 PATCH 524/626] powerpc: Remove non-uapi linkage.h export
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit dade934a5e62f99d3bd8377b2f401f559d827f6f
Author: James Hogan <james.hogan@imgtec.com>
Date:   Fri May 2 11:20:41 2014 +0100

    powerpc: Remove non-uapi linkage.h export

    The arch/powerpc/include/asm/linkage.h is being unintentionally exported
    in the kernel headers since commit e1b5bb6d1236 (consolidate
    cond_syscall and SYSCALL_ALIAS declarations) when
    arch/powerpc/include/uapi/asm/linkage.h was deleted but the header-y not
    removed from the Kbuild file. This happens because Makefile.headersinst
    still checks the old asm/ directory if the specified header doesn't
    exist in the uapi directory.

    The asm/linkage.h shouldn't ever have been exported anyway. No other
    arch does and it doesn't contain anything useful to userland, so remove
    the header-y line from the Kbuild file which triggers the export.

    Signed-off-by: James Hogan <james.hogan@imgtec.com>
    Cc: Benjamin Herrenschmidt <benh@kernel.crashing.org>
    Cc: Paul Mackerras <paulus@samba.org>
    Cc: linuxppc-dev@lists.ozlabs.org
    Cc: Al Viro <viro@zeniv.linux.org.uk>
    Cc: David Woodhouse <dwmw2@infradead.org>
    Cc: David Howells <dhowells@redhat.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/uapi/asm/Kbuild b/arch/powerpc/include/uapi/asm/Kbuild
index 48be855..7a3f795 100644
--- a/arch/powerpc/include/uapi/asm/Kbuild
+++ b/arch/powerpc/include/uapi/asm/Kbuild
@@ -15,7 +15,6 @@ header-y += ioctls.h
 header-y += ipcbuf.h
 header-y += kvm.h
 header-y += kvm_para.h
-header-y += linkage.h
 header-y += mman.h
 header-y += msgbuf.h
 header-y += nvram.h
-- 
1.7.1